apache/superset
# refactor(Shared_url_query): Fix shared  query URL access for SQL Lab users.

### SUMMARY
This update fixes the copy sharing to use the newly introduced permalink.

Fixes #31367

fix(sql-lab): fix the copy url to use the permalink api and allow query sharing for users with appropriate permissions.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
![1737369289283](https://github.com/user-attachments/assets/a4c0ec56-6642-48c8-b024-9d0b3e372e63)


### TESTING INSTRUCTIONS
1. Log in and save a query in SQL Lab.
2. Copy the query URL using the "Copy Query URL" option.
3. Share the URL with a different user (one with the `sqllab` role and access to the dataset).
4. Have the other user try to load the shared query URL.
5. Observe that the second user with sqllab_role can successfully load the shared query.

### ADDITIONAL INFORMATION

- [x] Has associated issue: Fixes #31367
